### PR TITLE
Fixed encrypted configuration and jsonConfig layout

### PR DESCRIPTION
**Pull request by klein0r**
- https://github.com/ioBroker/ioBroker.hm-rpc/issues/1170

### CODE REVIEW

**Comment by sh-hedge** [comment]
@foxriver76 Could you merge this pull request? 
Would be good to fix the connection issue when using username and pw....

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (foxriver76) [inline .vscode/extensions.json:1]: Could we add the vs-code stuff to `gitignore` if it is nothing special for this project?

**Review by foxriver76 — APPROVED** [review]

File: .vscode/extensions.json (hunk 1 above)
Comment (sh-hedge) [reply .vscode/extensions.json:1]: Don't know about this stuff... 
@klein0r should answer ;-)

Comment (foxriver76) [reply .vscode/extensions.json:1]: seems like vscode needs explicit mapping for the json schema so okay to keep it in for me.

**Pull request merged**